cmd/kubeadm/app/cmd/util/cmdutil.go

```diff
@@ -62,15 +62,16 @@ func ValidateExactArgNumber(args []string, supportedArgs []string) error {
 	return nil
 }
 
-// FindExistingKubeConfig returns the localtion of kubeconfig
-func FindExistingKubeConfig(file string) string {
-	// The user did provide a --kubeconfig flag. Respect that and threat it as an
-	// explicit path without building a DefaultClientConfigLoadingRules object.
-	if file != kubeadmconstants.GetAdminKubeConfigPath() {
-		return file
+// FindExistingKubeConfig returns the location of a kubeconfig file.
+// If the user passed the --kubeconfig flag respect it, otherwise search for existing
+// kubeconfig files in default locations.
+func FindExistingKubeConfig(flags *pflag.FlagSet) string {
+	if flags != nil && flags.Changed(options.KubeconfigPath) {
+		value, err := flags.GetString(options.KubeconfigPath)
+		if err == nil {
+			return value
+		}
 	}
-	// The user did not provide a --kubeconfig flag. Find a config in the standard
```

Comment:
Nit : I personally would keep this comment

Reply:
well this is sort of covered in the comment above the function now.